### PR TITLE
Sasl2 merge master

### PR DESCRIPTION
**Pull request by NelsonVides**

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_25](https://circleci.com/gh/esl/MongooseIM/191183) / elasticsearch_and_cassandra_mnesia / a68179bb42bdbe7803bd3a39cb0cbd30748fbe22
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4125/191183/elasticsearch_and_cassandra_mnesia.25.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4125/191183/elasticsearch_and_cassandra_mnesia.25.3-amd64/big/ct_run.test%40c1b117c58305.2023-09-12_14.29.40/index.html)
**OK: 369** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

File: src/c2s/mongoose_c2s.erl [hunk 1]
```diff
@@ -413,10 +412,24 @@ handle_sasl_step(StateData, {failure, NewSaslAcc, Result}, Retries) ->
 handle_sasl_step(StateData, {error, NewSaslAcc, Result}, Retries) ->
     handle_sasl_error(StateData, NewSaslAcc, Result, Retries).
 
+% -spec handle_sasl_success(data(), term()) -> fsm_res().
```
Comment (chrzaszcz) [inline src/c2s/mongoose_c2s.erl:415]: Do we need this commented-out code?

**Review by chrzaszcz — APPROVED** [review]
Looks good, I added a comment

File: src/c2s/mongoose_c2s.erl (hunk 1 above)
Comment (NelsonVides) [reply src/c2s/mongoose_c2s.erl]: Goddamit 😅

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_25](https://circleci.com/gh/esl/MongooseIM/191317) / elasticsearch_and_cassandra_mnesia / aae7e20bbe11726359ccd2f84c8b7169bdbe67e2
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4125/191317/elasticsearch_and_cassandra_mnesia.25.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4125/191317/elasticsearch_and_cassandra_mnesia.25.3-amd64/big/ct_run.test%40da4526215747.2023-09-13_06.36.22/index.html)
**OK: 369** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
👌

**Comment by Neustradamus** [comment]
@NelsonVides: Good job!

**Pull request merged**